### PR TITLE
Fix gen format fails with >10 species

### PR DESCRIPTION
**Pull request by bhourahine**
Solution for #529

### CODE REVIEW

File: prog/dftb+/lib_io/formatout.F90 [hunk 1]
```diff
@@ -189,34 +188,38 @@ subroutine writeGenFormat_fid(fd, coord, species, speciesName, latVec, origin, t
...
 
-    write(formatSpecies, 101) nSpecies
-    write(fd, formatSpecies) (trim(speciesName(ii)), ii = 1, nSpecies)
+    do ii = 1, nSpecies
+      write(formatSpecies, "('(A',I0,')')")len(trim(speciesName(ii)))+1
+      write(fd, formatSpecies, advance='NO')trim(speciesName(ii))
+    end do
+    write(fd,*)
 
+    write(formatCoordinates, 102) floor(log10(real(nSpecies)))+1
```
Comment (awvwgk) [inline prog/dftb+/lib_io/formatout.F90:209]: Please, don't mix `format` statements and format strings, this makes IO routines even harder to read.

File: prog/dftb+/lib_io/formatout.F90 [hunk 2]
```diff
@@ -146,14 +146,13 @@ subroutine writeGenFormat_fid(fd, coord, species, speciesName, latVec, origin, t
...
-    character(6) :: formatSpecies
+    character(mc) :: formatSpecies, formatCoordinates
     integer :: ii
     logical :: tFractional, tHelical, tPeriodic
     real(dp) :: invLatVec(3,3)
 
-100 format(I5," ",A2)
-101 format("(",I2.2,"A3)")
-102 format(I5,I2,3E20.10)
+101 format(1X,I0," ",A2)
```
Comment (awvwgk) [inline prog/dftb+/lib_io/formatout.F90:154]: Is there a specific need for the leading space? Also I would prefer to use either `1X` or `" "` for spaces to keep the format statements consistent.

File: prog/dftb+/lib_io/formatout.F90 [hunk 3]
```diff
@@ -189,34 +188,38 @@ subroutine writeGenFormat_fid(fd, coord, species, speciesName, latVec, origin, t
...
     else
-      write(fd, 100) nAtom, "C"
+      write(fd, 101) nAtom, "C"
     end if
 
-    write(formatSpecies, 101) nSpecies
-    write(fd, formatSpecies) (trim(speciesName(ii)), ii = 1, nSpecies)
+    do ii = 1, nSpecies
+      write(formatSpecies, "('(A',I0,')')")len(trim(speciesName(ii)))+1
+      write(fd, formatSpecies, advance='NO')trim(speciesName(ii))
```
[the remark this answers is not in the document]
Comment (bhourahine) [reply prog/dftb+/lib_io/formatout.F90:205]: A near miss, but '(1x,a)' will do it correctly.

[the remark this answers is not in the document]
Comment (awvwgk) [reply prog/dftb+/lib_io/formatout.F90:205]: Oh yeah, now I see, `len(trim(speciesName(ii)))+1` to sneak the space in was something I missed.

File: prog/dftb+/lib_io/formatout.F90 (hunk 1 above)
Comment (bhourahine) [reply prog/dftb+/lib_io/formatout.F90:209]: Point taken (was just sticking to the messy style of the existing routine).

File: prog/dftb+/lib_io/formatout.F90 (hunk 3 above)
[the remark this answers is not in the document]
Comment (bhourahine) [reply prog/dftb+/lib_io/formatout.F90:205]: (depending on view about leading or trailing spaces).

File: prog/dftb+/lib_io/formatout.F90 (hunk 2 above)
Comment (bhourahine) [reply prog/dftb+/lib_io/formatout.F90:154]: Not really, resulting file just looks neater to me. The above '(a)' discussion replicates this behaviour anyway.

**Pull request merged**